### PR TITLE
feat(inference): add composable science loss library for IFU fitting

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add new rubix.inference.losses module with:
  - masked_gaussian_nll for uncertainty-aware IFU fitting
  - huber_data_loss for robust residual handling
  - combine_loss_fns for weighted composition of multiple data terms
- export loss helpers via rubix.inference
- add focused unit tests for math correctness, input validation, and optimizer integration
- document the new loss layer in inference docs/workflows

## Validation
- pre-commit run on changed files passed (black, isort, pydoclint)
- compileall passed for new module and tests
- pytest invocation in conda-run sandbox did not return output in this session (same known sandbox behavior), so this PR includes explicit unit coverage and hook-clean changes

### CODE REVIEW

File: rubix/inference/losses.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    if weights is not None and weights.shape != target.shape:
+        raise ValueError("weights must have the same shape as target")
+
+    residual = prediction - target
+    abs_residual = jnp.abs(residual)
+    delta_arr = jnp.asarray(delta, dtype=residual.dtype)
+
+    quadratic = 0.5 * residual**2
+    linear = delta_arr * (abs_residual - 0.5 * delta_arr)
+    huber = jnp.where(abs_residual <= delta_arr, quadratic, linear)
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/losses.py:137]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/210#discussion_r3113202707)
